components/homme/src/share/edge_mod_base.F90

```diff
@@ -434,8 +434,19 @@ subroutine FreeEdgeBuffer(edge)
     deallocate(edge%putmap)
     deallocate(edge%getmap)
     deallocate(edge%reverse)
+
+    deallocate(edge%moveLength)
+    deallocate(edge%movePtr)
+
+    deallocate(edge%Srequest)
+    deallocate(edge%Rrequest)
+    deallocate(edge%status)
+
+
+
 #if (defined HORIZ_OPENMP)
 !$OMP END MASTER
+!$OMP BARRIER
```

Comment:
On Cori, I was getting strange errors in the NGGPS JW test case initialization on the second call to "make_c0" (so this only occurs if you have the code that DSS'es the tracers).   I think it was an error deallocating an already deallocated pointer, but I've deleted the log files so I'm not 100% sure anymore.  This extra barrier fixed the problem.  I think the problem might be that a thread can leave the first call to "make_c0" and enter the second call to "make_c0" while the master thread is still in the process of deallocating the buffers. 

I agree with your comment - the extra barrier should not be necessary.  It is in a non-performance critical part of the code (idealized test case identification) so not sure if we should just leave it it to be safe, to spend time tracking down the issue.  It might also have been a few bad runs and the problem went away coincidentally with adding the thread barriers.